--- xz-5.1.3alpha/src/liblzma/api/lzma/container.h.text~	2014-02-10 20:53:46.000000000 +0100
+++ xz-5.1.3alpha/src/liblzma/api/lzma/container.h	2014-02-11 11:58:23.406404399 +0100
@@ -43,7 +43,7 @@
 /*
  * Preset flags
  *
- * Currently only one flag is defined.
+ * Currently only two flags are defined.
  */
 
 /**
@@ -59,6 +59,12 @@
  */
 #define LZMA_PRESET_EXTREME       (UINT32_C(1) << 31)
 
+/**
+ * \brief       Text compression preset
+ *
+ * TODO
+ */
+#define LZMA_PRESET_TEXT          (UINT32_C(1) << 18)
 
 #ifdef LZMA_UNSTABLE /* Unstable API that may change. Use only for testing. */
 /**
--- xz-5.1.3alpha/src/liblzma/lzma/lzma_encoder_presets.c.text~	2014-02-10 20:53:46.000000000 +0100
+++ xz-5.1.3alpha/src/liblzma/lzma/lzma_encoder_presets.c	2014-02-11 11:58:23.406404399 +0100
@@ -18,7 +18,7 @@ lzma_lzma_preset(lzma_options_lzma *opti
 {
 	const uint32_t level = preset & LZMA_PRESET_LEVEL_MASK;
 	const uint32_t flags = preset & ~LZMA_PRESET_LEVEL_MASK;
-	const uint32_t supported_flags = LZMA_PRESET_EXTREME;
+	const uint32_t supported_flags = LZMA_PRESET_EXTREME | LZMA_PRESET_TEXT;
 
 	if (level > 9 || (flags & ~supported_flags))
 		return true;
@@ -59,5 +59,18 @@ lzma_lzma_preset(lzma_options_lzma *opti
 		}
 	}
 
+	if (flags & LZMA_PRESET_TEXT) {
+		if (level == 0)
+			options->lc = 3;
+		else
+			options->lc = 4;
+		options->lp = 0;
+		options->pb = 0;
+		options->mode = LZMA_MODE_NORMAL;
+		options->nice_len = 273;
+		options->mf = LZMA_MF_BT4;
+		options->depth = 512;
+	}
+
 	return false;
 }
--- xz-5.1.3alpha/src/xz/args.c.text~	2014-02-10 20:53:46.000000000 +0100
+++ xz-5.1.3alpha/src/xz/args.c	2014-02-11 12:12:51.916381380 +0100
@@ -141,6 +141,7 @@ parse_real(args_info *args, int argc, ch
 		OPT_INFO_MEMORY,
 		OPT_ROBOT,
 		OPT_FLUSH_TIMEOUT,
+		OPT_TEXT,
 	};
 
 	static const char short_opts[]
@@ -182,6 +183,8 @@ parse_real(args_info *args, int argc, ch
 		{ "extreme",      no_argument,       NULL,  'e' },
 		{ "fast",         no_argument,       NULL,  '0' },
 		{ "best",         no_argument,       NULL,  '9' },
+		{ "text",         no_argument,       NULL,  OPT_TEXT },
+
 
 		// Filters
 		{ "lzma1",        optional_argument, NULL,  OPT_LZMA1 },
@@ -490,6 +493,10 @@ parse_real(args_info *args, int argc, ch
 					optarg, 0, UINT64_MAX);
 			break;
 
+		case OPT_TEXT:
+			coder_set_text();
+			break;
+
 		default:
 			message_try_help();
 			tuklib_exit(E_ERROR, E_ERROR, false);
--- xz-5.1.3alpha/src/xz/coder.c.text~	2014-02-10 20:53:46.000000000 +0100
+++ xz-5.1.3alpha/src/xz/coder.c	2014-02-11 11:58:23.406404399 +0100
@@ -59,6 +59,8 @@ static lzma_mt mt_options = {
 };
 #endif
 
+/// TODO: Description..
+static bool preset_text = false;
 
 extern void
 coder_set_check(lzma_check new_check)
@@ -101,6 +103,15 @@ coder_set_extreme(void)
 	forget_filter_chain();
 	return;
 }
+
+
+extern void
+coder_set_text(void)
+{
+	preset_number |= LZMA_PRESET_TEXT;
+	forget_filter_chain();
+	return;
+}
 
 
 extern void
--- xz-5.1.3alpha/src/xz/coder.h.text~	2014-02-10 20:53:46.000000000 +0100
+++ xz-5.1.3alpha/src/xz/coder.h	2014-02-11 11:58:23.406404399 +0100
@@ -61,6 +61,9 @@ extern void coder_set_preset(uint32_t ne
 /// Enable extreme mode
 extern void coder_set_extreme(void);
 
+/// Enable text mode
+extern void coder_set_text(void);
+
 /// Add a filter to the custom filter chain
 extern void coder_add_filter(lzma_vli id, void *options);
 
--- xz-5.1.3alpha/tests/test_compress.sh.text~	2014-02-11 13:03:45.431535071 +0100
+++ xz-5.1.3alpha/tests/test_compress.sh	2014-02-11 15:33:40.012017357 +0100
@@ -103,6 +103,7 @@ do
 
 	# Don't test with empty arguments; it breaks some ancient
 	# proprietary /bin/sh versions due to $@ used in test_xz().
+	test_xz -0 --text
 	test_xz -1
 	test_xz -2
 	test_xz -3
