diff -up xz-5.4.3/src/liblzma/liblzma.pc.in.omv~ xz-5.4.3/src/liblzma/liblzma.pc.in
--- xz-5.4.3/src/liblzma/liblzma.pc.in.omv~	2023-06-07 15:19:56.807101982 +0200
+++ xz-5.4.3/src/liblzma/liblzma.pc.in	2023-06-07 15:20:28.831296177 +0200
@@ -14,6 +14,5 @@ Name: liblzma
 Description: General purpose data compression library
 URL: @PACKAGE_URL@
 Version: @PACKAGE_VERSION@
-Cflags: -I${includedir}
-Libs: -L${libdir} -llzma
+Libs: -llzma
 Libs.private: @PTHREAD_CFLAGS@ @LIBS@
